packages/flutter/lib/src/cupertino/menu_anchor.dart

```diff
@@ -0,0 +1,3200 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:math' as math;
+import 'dart:ui' as ui;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/gestures.dart';
+import 'package:flutter/physics.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/scheduler.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter/widgets.dart';
+
+import '../material/material_localizations.dart'
+    show MaterialLocalizations;
+import '../material/menu_anchor.dart'
```

Comment:
The circular dependency problem is something I was unsure of how to fix. I asked a (long winded) question in the Discord chat, but no one got back to me (see [here](https://discord.com/channels/608014603317936148/608021635370582020/1215690029418942514)). Would you suggest I try moving the MenuAnchor core to the Widgets library, or do you have a suggestion? I'm mostly concerned about compromising any plans you all have, but I wasn't sure who to ask.

Reply:
> Would you suggest I try moving the MenuAnchor core to the Widgets library

This might be an option, let me ask some folks about it and circle back.

Reply:
Sounds good! I'll see what parts I can shed off of the CupertinoMenuAnchor to make it smaller.

Reply:
Moving MenuAnchor and MenuController --> widgets SGT-everyone I spoke with! Is this something you are interested in doing in a separate change? If not, I can send a PR. :)

The only material-specific bit we would need to leave in material would be MenuAnchor.menuStyle. So this would probably look like a RawMenuAnchor class in widgets, with  MenuAnchor as a subclass in material so it can have the MenuStyle.

Reply:
That's exciting! I'd be happy to. I believe I tried in the past and there are a few tricky dependencies between the two (mostly related to _Submenu, which depends on _MenuAnchorState internals, material theming, and _kMenuTraversalShortcuts,) but I'll go ahead and get a PR in so that we can discuss. I'll be out-of-town this weekend, but will submit ASAP!